src/integrationTest/java/uk/gov/hmcts/reform/pdf/service/integration/GeneratedPDFContentV2Test.java

```diff
@@ -0,0 +1,73 @@
+package uk.gov.hmcts.reform.pdf.service.integration;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableMap;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.pdfbox.pdmodel.PDDocument;
+import org.pdfbox.util.PDFTextStripper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import uk.gov.hmcts.reform.pdf.service.domain.GeneratePdfRequest;
+import uk.gov.hmcts.reform.pdf.service.endpoint.v2.PDFGenerationEndpointV2;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+public class GeneratedPDFContentV2Test {
+
+    private static final String API_URL = "/pdf";
+    private static ObjectMapper objectMapper = new ObjectMapper();
+
+    @Autowired
+    private MockMvc webClient;
+
+    private static String textContentOf(byte[] pdfData) throws IOException {
+        PDDocument pdfDocument = PDDocument.load(new ByteArrayInputStream(pdfData));
+        try {
+            return new PDFTextStripper().getText(pdfDocument);
+        } finally {
+            pdfDocument.close();
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void shouldCreateExpectedPdfFromPlainHtmlTemplate() throws Exception {
+        assertThatGeneratedPdfContains("<html><body>Hello!</body></html>", Collections.EMPTY_MAP, "Hello!");
+    }
+
+    @Test
+    public void shouldCreateExpectedPdfFromPlainTwigTemplateAndPlaceholders() throws Exception {
```

Comment:
using underscores would improve readability imho.
Plus it will be consistent as you've used underscores in `GeneratePdfRequestTest` :)

Reply:
I've duplicated the contents of `GeneratePdfRequestTest` and applied necessary amendments for APIv2. Let's address this in next PR.